### DISCUSSION TITLE
Interaction#fetchChannel

### DISCUSSION THREAD

**Discussion by tguichaoua** [discordjs/discord.js#6272] [Ideas]
Getting the channel from an `Interaction` require a pretty long boilerplate code (see below). I propose to add some shorcut method to `Interaction`.

Maybe the name `fetch` is not appropriate since it doesn't fetch in all case.

I know the policy is to avoid trivial shorcut method, but this one could help improve code readability.
```ts
const channel =
    interaction.channel ??
    (interaction.channelId
        ? await (
              interaction.guild ??
              (interaction.guildId ? await interaction.client.guilds.fetch(interaction.guildId) : null)
          )?.channels.fetch(interaction.channelId)
        : null) ??
    null;

// With shortcut method
const channel = await interaction.fetchChannel();
```

**Comment by iCrawl** [comment]
Just fetch either way really.

```js
const guild = await interaction.client.guilds.fetch(interaction.guildId);
const channel = await interaction.client.channels.fetch(interaction.channelId);
```

`.fetch` won't make an API request if you already have it in cache by default.
